tests/dragonfly/connection_test.py

```diff
@@ -346,3 +347,51 @@ async def test_subscribe_in_pipeline(async_client: aioredis.Redis):
     res = await pipe.execute()
 
     assert res == ['one', ['subscribe', 'ch1', 1], 'two', ['subscribe', 'ch2', 2], 'three']
+
+"""
+This test makes sure that Dragonfly can receive blocks of pipelined commands even
+while a script is still executing. This is a dangerous scenario because both the dispatch fiber
+and the connection fiber are actively using the context. What is more, the script execution injects
+its own custom reply builder, which can't be used anywhere else, besides the lua script itself.
+"""
+
+async def test_parser_while_script_running(async_client, df_server):
+    S = """
+    for i=1,30 do
+        redis.call('MGET', 'k1', 'k2', 'k3')
+    end
+    """
+
+    P1 = """
+    MGET s1 s2 s3
+    EVALSHA {sha} 3 k1 k2 k3
+    """
+
+    P2 = """
+    MGET m1 m2 m3
+    MGET m4 m5 m6
+    MGET m7 m8 m9\n
+    """
+    P3 = """
+    PING
+    """ * 50 + "ECHO DONE\n"
+
+    c = aioredis.Redis(port=df_server.port)
+    sha = await c.script_load(S)
+
+    reader, writer = await asyncio.open_connection('localhost', df_server.port)
+
+    writer.write(P1.format(sha=sha).encode())
+    await writer.drain()
+
+    await asyncio.sleep(0.01)
+
+    writer.write(P2.encode())
+    writer.write(P3.encode())
+    await writer.drain()
+
+    out = await reader.readuntil(b"DONE")
+    print(out)
+
+    writer.close()
```

Comment:
I do not think we need to check or assert anything. This test covers a specific use-case that crashes DF in debug mode. It's interesting enough already